test/cli/test/cli-options-workspace.test.ts

```diff
@@ -0,0 +1,15 @@
+import { expect, test } from 'vitest'
+import { runVitestCli } from '../../test-utils'
+
+test('workspace project options are overriden by CLI options', async () => {
+  const { stderr, stdout } = await runVitestCli(
+    { nodeOptions: { cwd: './fixtures/browser-workspace' } },
+    '--run',
+    '--browser.provider=playwright',
+    '--browser.isolate=false',
+  )
```

Comment:
I didn't want to test the reported `browser.headless` option here, as it has different resolving logic on CI.  Reverting the fix makes these options fail so it should be equivalent.